docs/contributing/Building, Debugging, and Testing on Windows.md

```diff
@@ -0,0 +1,91 @@
+# Required Software
+
+1. [Visual Studio 2015 with Update 1](http://go.microsoft.com/fwlink/?LinkId=691129). _You need Update 1_.
+2. Visual Studio 2015 Extensibility Tools. If you already installed Visual Studio, choose "Modify" from the Programs and Features control panel, and check "Visual Studio Extensibility".
+
+# Getting the Code
+
+1. Clone https://github.com/dotnet/roslyn
+2. Run the "Developer Command Prompt for VS2015" from your start menu.
+3. Run `Restore.cmd` in the command prompt to restore NuGet packages.
+4. Due to [Issue #5876](https://github.com/dotnet/roslyn/issues/5876), you should build on the command line before opening in Visual Studio. Run `msbuild /v:m /m Roslyn.sln`
```

Comment:
Another option is to just use the same hook we use in cibuild.cmd to take over the compiler.  CIBuild essentially does the following:

- Build toolset.sln
- Copy `Binaries\Debug` to a new directory `Binaries\Bootstrap`
- Run msbuild with `/p:BootstrapBuildPath=Binaries\Bootstrap`

The `BootstrapBuildPath` hook isn't specific to any path.  You could point it to any directory on your machine so long as it had csc and vbc inside it.  

> I agree that's not discoverable

It's not discoverable because it generally isn't deemed valuable to individual developers.  Even on the compiler team we use the toolset compiler for the vast majority of our development.  The reason for cibuild using the freshly built compiler is just a sanity check that we can always compile our repo with our repo.  

This is a valuable test to have but the breaks it catches are rather rare.  Typically we get about 1 per month on the compiler team.  The decreased throughput of the compiler when using bootstrap builds isn't worth catching a bug that you'd introduce on average once a year.  Hence we do it only in our build lab.